### ISSUE TITLE
Features required for `--fix` support 

### ISSUE THREAD

**Issue by rwjblue** [ember-template-lint/ember-template-lint#1180]
- [x] Ensure file is updated when passing `--fix` (https://github.com/ember-template-lint/ember-template-lint/pull/1183)
- [x] Update the pretty formatter to emit slightly different output when fixable errors are detected (like eslint does) https://github.com/ember-template-lint/ember-template-lint/pull/1185:

```
  1 error and 0 warnings potentially fixable with the `--fix` option.
```

- [x] Add end to end acceptance tests for fixing (that invoke `ember-template-lint` with a template that needs fixes, and confirms the updated output) (https://github.com/ember-template-lint/ember-template-lint/pull/1183)

- [x] Ensure file with BOM is written back to disk with BOM #1190 

- [ ] Write text confirming functionality when reading from STDIN (needs some research, Do we write to STDOUT? What does ALE expect? What does eslint do here?)

**Comment by dcyriller** [comment]
Just did some eslint research, I'm writing the results here for future reference 😄.

`eslint` has the following options: `--format=json`, `--stdin`, `--stdin-filename` (we have similar concepts in ember-template-lint).

Aside from these options, there is `--fix-dry-run`, which is an interesting option and could be ported pretty easily.

There is also a `--fix-type` option to let users specify which kind of fixes should be applied (`problem` / `suggestion` / `layout`).

I also see that with older versions of eslint, Ale would run `yarn eslint --fix the-file.js` ^^.

**Comment by locks** [comment]
Most tasks listed are complete, and there is a separate issue for the last task regarding STDOUT, so I'm closing this one. Thanks for the work!
